pkg/services/ngalert/api/tooling/definitions/alertmanager.go

```diff
@@ -567,8 +567,9 @@ func (c *PostableUserConfig) UnmarshalYAML(value *yaml.Node) error {
 
 // swagger:model
 type GettableUserConfig struct {
-	TemplateFiles      map[string]string         `yaml:"template_files" json:"template_files"`
-	AlertmanagerConfig GettableApiAlertingConfig `yaml:"alertmanager_config" json:"alertmanager_config"`
+	TemplateFiles           map[string]string            `yaml:"template_files" json:"template_files"`
+	TemplateFileProvenances map[string]models.Provenance `yaml:"template_file_provenances,omitempty" json:"template_file_provenances,omitempty"`
```

Comment:
I wonder if we can make it `map[string]TemplateFile` where template file would be a struct 
```
type TemplateFile struct {
    Content string
    Provenance model.Provenance. 
}
```

this will require simulataneous changes in UI but I think that would look cleaner.

Reply:
I agree it's cleaner, but I intentionally avoided this to not introduce a breaking change. Is the UI the only thing we're worried about breaking at this point?

Reply:
If I am not mistaken the API is still not GA, so we can change it at our own discretion without deprecating\versioning things.

Reply:
Let's do this after 9.0. This will need some more consideration as we then cannot use the editor anymore to just paste configs around.